site/templates/message.html

```diff
@@ -53,122 +56,134 @@ <h2>Send Brian a message on his Tidbyt.</h2>
       </button>
     </div>
-
   </div>
 
-  <div class="row justify-content-center" id="response_messages"></div>
+  <div class="row justify-content-center" id="stop-response-div"></div>
 </div>
 
 <script>
-  function removeAllChildNodes(parent) {
-    while (parent.firstChild) {
-      parent.removeChild(parent.firstChild);
-    }
-  }
-
-  // toggle button
-  async function toggleButton(buttonId, spinnerSpanId, enabledText, disabledText) {
-    const btn = document.getElementById(buttonId);
-    const btnSpan = document.getElementById(spinnerSpanId);
-
-    const btnContent = btn.textContent;
-
-    if (btnContent.includes(enabledText)) {
-      btn.setAttribute("disabled", "");
-      btnSpan.removeAttribute("style");
-      const inner = btn.innerHTML;
-      const newInner = inner.replace(enabledText, disabledText);
-      btn.innerHTML = newInner;
-    } else if (btnContent.includes(disabledText)) {
-      btn.removeAttribute("disabled");
-      btnSpan.setAttribute("style", "display: none");
-      const inner = btn.innerHTML;
-      const newInner = inner.replace(disabledText, enabledText);
-      btn.innerHTML = newInner;
-    }
-  }
+  const IS_MOCKING = true;
+  const mockDelay = (seconds) => new Promise((resolve) => setTimeout(resolve, seconds * 1000));
+
+  const API_URL_ROOT = "https://api.brianwtracey.com";
+  const apiFetch = async (path, options) => await fetch(`${API_URL_ROOT}/${path}`, options);
+  const api = {
+    stop: async () => {
+      if (IS_MOCKING) {
+        await mockDelay(2);
+        return "Thanks for telling Brian to stop talking.";
+      }
+
+      const response = await apiFetch("stop");
+      return await response.json();
+    },
+    message: async (body) => {
+      if (IS_MOCKING) {
+        await mockDelay(2);
+        return { status: "success" };
+      }
+
+      const response = await apiFetch("message", {
+        method: "post",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(body),
+      });
+      return await response.json();
+    },
+  };
+
+  const elements = {
+    message: { button: document.getElementById("submit-button") },
+    stop: { button: document.getElementById("stop-button") },
+  };
+  elements.message.form = document.getElementById("message-form");
+  elements.message.spinner = elements.message.button.getElementsByClassName("spinner-grow");
+  elements.message.text = elements.message.button.getElementsByClassName("text");
+  elements.message.response = document.getElementById("message-response-div");
+  elements.stop.spinner = elements.stop.button.getElementsByClassName("spinner-grow");
+  elements.stop.text = elements.stop.button.getElementsByClassName("text");
+  elements.stop.response = document.getElementById("stop-response-div");
+
+  const strings = {
+    message: "Send Message",
+    stop: "Tell Brian to stop talking.",
+    sending: "Sending...",
```

Comment:
Like the elements, if I work with static strings at all, I keep them in an object so I don't have to worry about mistyping. Plus it becomes very easy to compare things in the future. Or if you ever want to change messages, you just change it in one place.